scylla_phot_culls.py

```diff
@@ -0,0 +1,183 @@
+            continue
+        inds_to_cut = np.where((t[filters[i] + '_CROWD'] > crowdcut[i]) |
+                               (t[filters[i] + '_SHARP'] * t[filters[i] + '_SHARP'] > sharpcut[i]) |
+                               (t[filters[i] + '_ROUND'] > roundcut[i]))[0]
+        if errcut:
+            errinds_to_cut = np.where(t[filters[i] + '_ERR'] > errcut)[0]
+            inds_to_cut = np.unique(np.concatenate((inds_to_cut, errinds_to_cut), 0))
+
+        t[filters[i] + '_VEGA'][inds_to_cut] = 99.999
+        t[filters[i] + '_STD'][inds_to_cut] = 99.999
+        t[filters[i] + '_ERR'][inds_to_cut] = 99.999
+        t[filters[i] + '_CHI'][inds_to_cut] = 99.999
+        t[filters[i] + '_SNR'][inds_to_cut] = 99.999
+        t[filters[i] + '_SHARP'][inds_to_cut] = 99.999
+        t[filters[i] + '_ROUND'][inds_to_cut] = 99.999
+        t[filters[i] + '_CROWD'][inds_to_cut] = 99.999
+        t[filters[i] + '_FLAG'][inds_to_cut] = 99
+
+    t.write(catalog_out, overwrite=True)
+    print('Done culling in %.2f sec.' % round((time.time()-start_time_cull), 2))
+    print('New catalog:' % catalog_out)
+
+    if plot:
+
+        inds_noncut = np.where((t[filters[0] + '_FLAG'] < 99) & (t[filters[1] + '_FLAG'] < 99))[0]
+        ra = t['RA'][inds_noncut]
+        dec = t['DEC'][inds_noncut]
+
+        if errcut:
+            errcutstr = '_errcut'
+        else:
+            errcutstr = ''
+
+        fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(9.9, 4.5))
+        fig.suptitle('Scylla %s%s (vgst catalog)' % (field_id, errcutstr))
+        ax1.plot(ra, dec, ',', ls='', label='N = %s' % len(ra))
+        ax1.set_xlabel('RA', fontsize=15)
+        ax1.set_ylabel('DEC', fontsize=15)
+        ax1.legend(loc='upper right')
+
+        # only plot CMDs if multiple filters are provided
+        if len(filters) > 1:
+            inds_non99 = np.where((t[filters[0] + '_VEGA'] < 99) & (t[filters[1] + '_VEGA'] < 99))[0]
+            inds_cmd = np.intersect1d(inds_noncut, inds_non99)
+            mag1 = t[filters[0] + '_VEGA'][inds_cmd]
+            mag2 = t[filters[1] + '_VEGA'][inds_cmd]
+
+            # Set up CMD Kernel Density Estimation (KDE) scatter plot to see the density of sources
+            print('Setting up KDE scatter plot...')
+            start_time_kde = time.time()
+            xy = np.vstack([(mag1-mag2), mag1])
+            z = gaussian_kde(xy)(xy)
+            idx = z.argsort()
+            x, y, z = (mag1-mag2)[idx], mag1[idx], z[idx]
+            print('Scatter plot set up in %.2f sec.' % round((time.time()-start_time_kde), 2))
+            ax2.scatter(x, y, c=z, s=3, cmap='GnBu_r', label='N = %s' % len(mag1))
+
+            ax2.set_xlabel('%s - %s' % (filters[0], filters[1]), fontsize=15)
+            ax2.set_ylabel('%s' % filters[0], fontsize=15)
+            ax2.set_xlim(-2, 5)
+            ax2.set_ylim(32.5, min(mag1)-0.5)
+            ax2.legend(loc='upper right')
+
+        plt.savefig('scylla-%s%s.png' % (field_id, errcutstr))
```

Comment:
Would it be possible to save the png in the same spot at the `catalog_out` so that the two stay connected. 
Could be done with code like
`plt.savefig(catalog_out.replace(".fits", "%s.png" %errcutstr))`
`print('Plot saved as %s' catalog_out.replace(".fits", "%s.png" %errcutstr))`

Reply:
Makes sense, implemented this.